src/Service/PerformedOperations.php

```diff
@@ -0,0 +1,241 @@
+   *   This posts contains an array of posts objects.
+   *
+   * @return array
+   *  Returning array of posts data.
+   */
+  public function postList(array $posts)
+  {
+    $postList = [];
+    // Iterating the users list to individual users list.
+    foreach ($posts as $post) {
+      $postList[] = [
+        'userImage' => $post->getUser()->getImageName(),
+        'userName'  => $post->getUser()->getFullName(),
+        'commentNo' => count($this->comments($post)),
+        'userId'    => $post->getUser()->getId(),
+        'createdAt' => $post->getCreatedAt(),
+        'updatedAt' => $post->getUpdatedAt(),
+        'content'   => $post->getContent(),
+        'likes'     => $this->likes($post),
+        'postImage' => $post->getImage(),
+        'postLikes' => $post->getLikes(),
+        'postId'    => $post->getId(),
+      ];
+    }
+    return $postList;
+  }
+  /**
+   * This function creates a array for storing user data.
+   * 
+   * @param array $users
+   *   This function will be used to fetch different user data and store in
+   *   an array.
+   *
+   * @return array
+   *  Returning array of users data.
+   */
+  public function getUserData(array $users)
+  {
+    $userList = [];
+    // Iterating the users list to individual users list.
+    foreach ($users as $user) {
+      $userList[] = [
+        'lastActiveTime' => $user->getLastActiveTime(),
+        'img'            => $user->getImageName(),
+        'fullName'       => $user->getFullName(),
+        'userId'         => $user->getId()
+      ];
+    }
+    return $userList;
+  }
+  /**
+   * Sanitize data is used to validate user inserted input if user has inserted
+   * any malicious value ini the input field, this function will not allow it to
+   * execute.
+   *
+   * @param string $data
+   *   This data is the string that will be sanitized.
+   * 
+   * @return string
+   *   This string data is the sanitized string data.
+   */
+  public function sanitizeData($data)
+  {
+    return htmlspecialchars($data, ENT_QUOTES, 'UTF-8');
+  }
+
+  /**
+   * This like returns the users who have liked a post.
+   *
+   * @param object $post
+   *   This post is the a post which will have likes and will be sent to the 
+   *   client side.
+   * 
+   * @return array
+   *   An array of users who have liked the post.
+   */
+  public function likes(object $post)
+  {
+    $likes = [];
+    foreach ($post->getLikes() as $like) {
+      $likes[] = [
+        "likes" => $like->getUser()->getEmail()
+      ];
+    }
+    return $likes;
+  }
+  /**
+   * This block of functions extracts the comments and stored in the array.
+   *
+   * @param object $post
+   *   This post is the a post which will have comments and will be sent to the 
+   *   client side.
+   * 
+   * @return array
+   *   An array of comments of the post.
+   */
+  public function comments(object $post)
+  {
+    $comments = [];
+    foreach ($post->getComments() as $comment) {
+      $comments[] = [
```

Comment:
Dont need to add the square brackets after comments

Reply:
Yes, you are right.